controllers/chat/sendSignedMessage.js

```diff
@@ -0,0 +1,79 @@
+const Validator = require('fastest-validator');
+const {StreamChat} = require('stream-chat');
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+
+const v = new Validator();
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const sendSignedMesage = async (req, res) => {
+  const schema = {
+    channelId: 'string|empty:false',
+    message: 'string|empty:false',
+    channelType: 'number|empty:false'
+  };
+  const validated = v.validate(req.body, schema);
+  if (validated.length)
+    return res.status(403).json({
+      message: 'Error validation',
+      error: validated
+    });
+
+  const {channelId, message, channelType} = req.body;
+
+  let channelTypeDef;
+
+  switch (channelType) {
+    case CHANNEL_TYPE.CHAT:
+    case CHANNEL_TYPE.ANONYMOUS:
+      channelTypeDef = 'messaging';
+      break;
+
+    case CHANNEL_TYPE.GROUP:
+      channelTypeDef = 'group';
+      break;
+
+    default:
+      break;
+  }
+
+  if (!channelTypeDef)
+    return res.status(403).json({
+      message: 'Error validation',
+      error: 'Channel type not found'
+    });
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    await client.connectUser({id: req.userId}, req.token);
+
+    const channel = client.channel(channelTypeDef, channelId);
+
+    const createdChannel = await channel.create();
+
+    if (createdChannel?.channel?.is_channel_blocked)
+      return res.status(403).json(responseError('Channel is blocked'));
+
+    const chat = await channel.sendMessage({
+      user_id: req.userId,
+      text: message,
+      ...req.body
+    });
+
+    await client.disconnectUser();
+    return res.status(200).json(responseSuccess('sent', chat));
+  } catch (error) {
+    await client.disconnectUser();
+    return res.status(error.statusCode ?? error.status ?? 400).json({
+      status: 'error',
+      code: error.statusCode ?? error.status ?? 400,
+      message: error.message
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in the try-catch block is well done. It disconnects the user from the client in both success and error scenarios which is a good practice to free up resources. The error response also includes the status code and the error message which is helpful for debugging. However, it's better to have a default status code of 500 (Internal Server Error) instead of 400 (Bad Request) in case the error object doesn't have a `statusCode` or `status`.

```diff
- return res.status(error.statusCode ?? error.status ?? 400).json({
+ return res.status(error.statusCode ?? error.status ?? 500).json({
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->